api/plugin.go

```diff
@@ -0,0 +1,470 @@
+		}
+	})
+}
+
+// GetConfig returns Configurer plugin configuration in YAML format.
+// swagger:operation GET /plugin/:id/config plugin getPluginConfig
+//
+// Get YAML configuration for Configurer plugin.
+//
+// ---
+// consumes: [application/json]
+// produces: [application/x-yaml]
+// security: [clientTokenHeader: [], clientTokenQuery: [], basicAuth: []]
+// responses:
+//   200:
+//     description: Ok
+//     schema:
+//         type: object
+//         description: plugin configuration
+//   404:
+//     description: Not Found
+//     schema:
+//         $ref: "#/definitions/Error"
+//   500:
+//     description: Internal Server Error
+//     schema:
+//         $ref: "#/definitions/Error"
+//   501:
+//     description: Not Implemented
+//     schema:
+//         $ref: "#/definitions/Error"
+func (c *PluginAPI) GetConfig(ctx *gin.Context) {
+	withID(ctx, "id", func(id uint) {
+		plugin := c.DB.GetPluginConfByID(id)
+		if plugin == nil {
+			ctx.AbortWithError(404, errors.New("unknown plugin"))
+			return
+		}
+		instance, err := c.Holder.GetInstance(id)
+		if err != nil {
+			ctx.AbortWithError(500, errors.New("plugin instance not found"))
+			return
+		}
+		if conf := instance.GetConfigurer(); conf != nil {
+			ctx.Header("content-type", "application/x-yaml")
+			ctx.Writer.Write(plugin.Config)
+		} else {
+			ctx.AbortWithError(501, errors.New("plugin does not implement Configurer"))
+		}
+	})
+
+}
+
+// UpdateConfig updates Configurer plugin configuration in YAML format.
+// swagger:operation POST /plugin/:id/config plugin updatePluginConfig
+//
+// Update YAML configuration for Configurer plugin.
+//
+// ---
+// consumes: [application/x-yaml]
+// produces: [application/json]
+// security: [clientTokenHeader: [], clientTokenQuery: [], basicAuth: []]
+// responses:
+//   200:
+//     description: Ok
+//   400:
+//     description: Bad Request
+//     schema:
+//         $ref: "#/definitions/Error"
+//   404:
+//     description: Not Found
+//     schema:
+//         $ref: "#/definitions/Error"
+//   500:
+//     description: Internal Server Error
+//     schema:
+//         $ref: "#/definitions/Error"
+//   501:
+//     description: Not Implemented
+//     schema:
+//         $ref: "#/definitions/Error"
+func (c *PluginAPI) UpdateConfig(ctx *gin.Context) {
+	withID(ctx, "id", func(id uint) {
+		if conf := c.DB.GetPluginConfByID(id); conf == nil {
+			ctx.AbortWithError(404, errors.New("unknown plugin"))
+			return
+		}
+		instance, err := c.Holder.GetInstance(id)
+		if err != nil {
+			ctx.AbortWithError(500, errors.New("plugin instance not found"))
+			return
+		}
+		if con := instance.GetConfigurer(); con != nil {
+			newConf := con.EmptyConfig()
+			conf, err := ioutil.ReadAll(ctx.Request.Body)
+			if err != nil {
+				ctx.AbortWithError(500, err)
+			}
+			if err := yaml.Unmarshal(conf, newConf); err != nil {
+				ctx.AbortWithError(400, err)
+			}
+			if err := con.ValidateAndSetConfig(newConf); err != nil {
```

Comment:
When this is successful, the config should be saved inside the pluginConf. 

Also, it would be good to also include this functionality to the plugin.manager (then all access of the plugin conf would be encapsulated by the plugin.Manager).
```go
func (m *Manager) SaveConfig(pluginID uint, config string) (statusCode int, err error)
```

Reply:
The DB update has been added in https://github.com/gotify/server/pull/110/commits/f86e57b316c1e18a116d1d7f1c2dd3544b22e675

I don't really like the idea of handling the unmarshaling work to the manager, it does not make things less coupled because the returned `statusCode` is another form of coupling.